x-pack/test/api_integration/deployment_agnostic/services/alerting_api.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RoleCredentials } from '@kbn/ftr-common-functional-services';
+import { DeploymentAgnosticFtrProviderContext } from '../ftr_provider_context';
+
+export function AlertingApiProvider({ getService }: DeploymentAgnosticFtrProviderContext) {
+  const retry = getService('retry');
+  const samlAuth = getService('samlAuth');
+  const supertest = getService('supertest');
+  const es = getService('es');
+  const requestTimeout = 30 * 1000;
+  const retryTimeout = 120 * 1000;
+  const logger = getService('log');
+
+  return {
+    async waitForRuleStatus({
+      ruleId,
+      expectedStatus,
+      roleAuthc,
+    }: {
+      ruleId: string;
+      expectedStatus: string;
+      roleAuthc: RoleCredentials;
+    }) {
+      if (!ruleId) {
+        throw new Error(`'ruleId' is undefined`);
+      }
+      return await retry.tryForTime(retryTimeout, async () => {
+        const response = await supertest
+          .get(`/api/alerting/rule/${ruleId}`)
+          .set(roleAuthc.apiKeyHeader)
```

Comment:
Same here. Please use only `supertestWithoutAuth` to make API call with api key for the specified role

Reply:
I think this service can be used as an example to make quick changes https://github.com/elastic/kibana/blob/main/x-pack/test/api_integration/deployment_agnostic/services/slo_api.ts